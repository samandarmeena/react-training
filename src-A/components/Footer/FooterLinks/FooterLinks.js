import React from 'react';
import PropTypes from 'prop-types';
import styles from './FooterLinks.module.scss';
import Text from '../../Shared/Text';

/**
 * FooterLinks component
 * renders Footer links container
 * @param {Array} footerLinkStack
 * @return Element
 */
const FooterLinks = (props) => {
  const { footerLinkStack } = props;
  return (
    <>
      {footerLinkStack.map((footerLinks, index) => {
        return (
          <div key={index} className={styles.footerLinksColumn}>
            {footerLinks.map((link, key) => {
              return (
                <div key={key} className={styles.FooterLink}>
                  <Text
                    text={link.headerTitle ? link.headerTitle : link.linkTitle}
                    isStrong={link.headerTitle ? true : false}
                  />
                </div>
              );
            })}
          </div>
        );
      })}
    </>
  );
};

// proptype validations for props
FooterLinks.propTypes = {
  footerLinkStack: PropTypes.array.isRequired
};

export default FooterLinks;
