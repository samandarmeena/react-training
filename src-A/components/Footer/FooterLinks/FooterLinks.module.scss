.footerLinksColumn {
  width: 25%;
  .FooterLink {
    margin: 10px auto;
  }
}
