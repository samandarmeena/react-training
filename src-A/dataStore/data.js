import { IMAGE_URL } from '../constants/constants';
// Footer links array for FooterLinks component
const footerLinkStack = [
  [
    { headerTitle: 'Customer Service' },
    { linkTitle: 'Help Center' },
    { linkTitle: 'Contact Us' },
    { linkTitle: 'Return an Item' },
    { linkTitle: 'Recall Information' },
    { linkTitle: 'Safety Data Sheets' }
  ],
  [
    { headerTitle: 'Corporate Info' },
    { linkTitle: 'Staples Business Advantage Overview' },
    { linkTitle: 'Industries Served' },
    { linkTitle: 'Blog' },
    { linkTitle: 'Our Brands' }
  ],
  [
    { headerTitle: 'Staples Corporate Solutions' },
    { linkTitle: 'Office Products' },
    { linkTitle: 'Technology Products' },
    { linkTitle: 'Furniture' },
    { linkTitle: 'Facilities' },
    { linkTitle: 'Promotional Products' }
  ],
  [
    { headerTitle: 'New Customer' },
    { linkTitle: 'Become a Customer' },
    { linkTitle: 'Govt. Customers' },
    { headerTitle: 'Memberships' },
    { linkTitle: 'Join the Staples Team' },
    { linkTitle: 'Careers' }
  ]
];

// accessibility image for FooterConnect component
const accessibilityImg = {
  src: `${IMAGE_URL}essential-accessibility.png`,
  alt: 'Essential Accessibility'
};

// social connect image for FooterConnect component
const socialImgs = [
  { src: `${IMAGE_URL}linkedin.png`, alt: 'LinkedIn' },
  { src: `${IMAGE_URL}twitter.png`, alt: 'Twitter' },
  { src: `${IMAGE_URL}facebook.png`, alt: 'Facebook' },
  { src: `${IMAGE_URL}youtube.png`, alt: 'Youtube' },
  { src: `${IMAGE_URL}insta.png`, alt: 'Instagram' }
];

// copyright text for footer copyright
const copyrightText =
  'Copyright &copy; 2001-2019 Corporate Express Canada, Inc., a Staples company. All rights reserved.';

// copyright links for footer copyright
const copyRightLinks = [
  'Terms & Conditions',
  ' | ',
  'Privacy Policy',
  ' | ',
  'Staples Bussiness Advantage'
];

// export all data
export {
  footerLinkStack,
  accessibilityImg,
  socialImgs,
  copyRightLinks,
  copyrightText
};
