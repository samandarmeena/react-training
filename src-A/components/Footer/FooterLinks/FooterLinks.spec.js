import React from 'react';
import { mount } from 'enzyme';
import FooterLinks from './FooterLinks';
import { footerLinkStack } from '../../../dataStore/data';

describe('FooterLink component', () => {
  let wrapper = '';
  beforeAll(() => {
    wrapper = mount(<FooterLinks footerLinkStack={footerLinkStack} />);
  });

  it('component rendered without an error', () => {
    expect(wrapper).toBeTruthy();
  });

  it('Text is rendering properly', () => {
    expect(
      wrapper
        .find('span')
        .first()
        .text()
    ).toBe('Customer Service');
    expect(
      wrapper
        .find('span')
        .at(16)
        .text()
    ).toBe('Promotional Products');
  });

  it('Check object keys', () => {
    expect(
      wrapper
        .find('Text')
        .first()
        .prop('isStrong')
    ).toBe(true);

    expect(
      wrapper
        .find('Text')
        .at(1)
        .prop('isStrong')
    ).toBe(false);
  });

  it('Check dynamically added class', () => {
    expect(
      wrapper
        .find('span')
        .at(2)
        .hasClass('textStrong')
    ).toBe(false);
    expect(
      wrapper
        .find('span')
        .at(6)
        .hasClass('textStrong')
    ).toBe(true);
  });

  it('Snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
