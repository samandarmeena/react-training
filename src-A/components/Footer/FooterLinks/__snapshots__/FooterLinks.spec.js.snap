// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLink component Snapshot testing 1`] = `
<FooterLinks
  footerLinkStack={
    Array [
      Array [
        Object {
          "headerTitle": "Customer Service",
        },
        Object {
          "linkTitle": "Help Center",
        },
        Object {
          "linkTitle": "Contact Us",
        },
        Object {
          "linkTitle": "Return an Item",
        },
        Object {
          "linkTitle": "Recall Information",
        },
        Object {
          "linkTitle": "Safety Data Sheets",
        },
      ],
      Array [
        Object {
          "headerTitle": "Corporate Info",
        },
        Object {
          "linkTitle": "Staples Business Advantage Overview",
        },
        Object {
          "linkTitle": "Industries Served",
        },
        Object {
          "linkTitle": "Blog",
        },
        Object {
          "linkTitle": "Our Brands",
        },
      ],
      Array [
        Object {
          "headerTitle": "Staples Corporate Solutions",
        },
        Object {
          "linkTitle": "Office Products",
        },
        Object {
          "linkTitle": "Technology Products",
        },
        Object {
          "linkTitle": "Furniture",
        },
        Object {
          "linkTitle": "Facilities",
        },
        Object {
          "linkTitle": "Promotional Products",
        },
      ],
      Array [
        Object {
          "headerTitle": "New Customer",
        },
        Object {
          "linkTitle": "Become a Customer",
        },
        Object {
          "linkTitle": "Govt. Customers",
        },
        Object {
          "headerTitle": "Memberships",
        },
        Object {
          "linkTitle": "Join the Staples Team",
        },
        Object {
          "linkTitle": "Careers",
        },
      ],
    ]
  }
>
  <div
    className="footerLinksColumn"
    key="0"
  >
    <div
      className="FooterLink"
      key="0"
    >
      <Text
        isStrong={true}
        text="Customer Service"
      >
        <span
          className="textStrong"
        >
          Customer Service
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="1"
    >
      <Text
        isStrong={false}
        text="Help Center"
      >
        <span
          className=""
        >
          Help Center
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="2"
    >
      <Text
        isStrong={false}
        text="Contact Us"
      >
        <span
          className=""
        >
          Contact Us
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="3"
    >
      <Text
        isStrong={false}
        text="Return an Item"
      >
        <span
          className=""
        >
          Return an Item
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="4"
    >
      <Text
        isStrong={false}
        text="Recall Information"
      >
        <span
          className=""
        >
          Recall Information
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="5"
    >
      <Text
        isStrong={false}
        text="Safety Data Sheets"
      >
        <span
          className=""
        >
          Safety Data Sheets
        </span>
      </Text>
    </div>
  </div>
  <div
    className="footerLinksColumn"
    key="1"
  >
    <div
      className="FooterLink"
      key="0"
    >
      <Text
        isStrong={true}
        text="Corporate Info"
      >
        <span
          className="textStrong"
        >
          Corporate Info
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="1"
    >
      <Text
        isStrong={false}
        text="Staples Business Advantage Overview"
      >
        <span
          className=""
        >
          Staples Business Advantage Overview
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="2"
    >
      <Text
        isStrong={false}
        text="Industries Served"
      >
        <span
          className=""
        >
          Industries Served
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="3"
    >
      <Text
        isStrong={false}
        text="Blog"
      >
        <span
          className=""
        >
          Blog
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="4"
    >
      <Text
        isStrong={false}
        text="Our Brands"
      >
        <span
          className=""
        >
          Our Brands
        </span>
      </Text>
    </div>
  </div>
  <div
    className="footerLinksColumn"
    key="2"
  >
    <div
      className="FooterLink"
      key="0"
    >
      <Text
        isStrong={true}
        text="Staples Corporate Solutions"
      >
        <span
          className="textStrong"
        >
          Staples Corporate Solutions
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="1"
    >
      <Text
        isStrong={false}
        text="Office Products"
      >
        <span
          className=""
        >
          Office Products
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="2"
    >
      <Text
        isStrong={false}
        text="Technology Products"
      >
        <span
          className=""
        >
          Technology Products
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="3"
    >
      <Text
        isStrong={false}
        text="Furniture"
      >
        <span
          className=""
        >
          Furniture
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="4"
    >
      <Text
        isStrong={false}
        text="Facilities"
      >
        <span
          className=""
        >
          Facilities
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="5"
    >
      <Text
        isStrong={false}
        text="Promotional Products"
      >
        <span
          className=""
        >
          Promotional Products
        </span>
      </Text>
    </div>
  </div>
  <div
    className="footerLinksColumn"
    key="3"
  >
    <div
      className="FooterLink"
      key="0"
    >
      <Text
        isStrong={true}
        text="New Customer"
      >
        <span
          className="textStrong"
        >
          New Customer
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="1"
    >
      <Text
        isStrong={false}
        text="Become a Customer"
      >
        <span
          className=""
        >
          Become a Customer
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="2"
    >
      <Text
        isStrong={false}
        text="Govt. Customers"
      >
        <span
          className=""
        >
          Govt. Customers
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="3"
    >
      <Text
        isStrong={true}
        text="Memberships"
      >
        <span
          className="textStrong"
        >
          Memberships
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="4"
    >
      <Text
        isStrong={false}
        text="Join the Staples Team"
      >
        <span
          className=""
        >
          Join the Staples Team
        </span>
      </Text>
    </div>
    <div
      className="FooterLink"
      key="5"
    >
      <Text
        isStrong={false}
        text="Careers"
      >
        <span
          className=""
        >
          Careers
        </span>
      </Text>
    </div>
  </div>
</FooterLinks>
`;
